examples/ex39.cpp

```diff
@@ -0,0 +1,287 @@
+//                                MFEM Example 39
+//
+// Compile with: make ex39
+//
+// Sample runs:  ex39
+//               ex39 -ess "Southern Boundary"
+//               ex39 -src Base
+//
+// Device sample runs:
+//               ex39 -fa -d cuda
+//
+// Description:  This example code demonstrates the use of named attribute
+//               sets in MFEM to specify material regions, boundary regions,
+//               or source regions by name rather than attribute numbers. It
+//               also demonstrates how new named attribute sets may be created
+//               from arbitrary groupings of attribute number and used as a
+//               convenient shorthand to refer to those groupings in other
+//               portions of the application or through the command line.
+//
+//               The particular problem being solved here is nearly the same
+//               as that in example 1 i.e. a simple finite element
+//               discretization of the Laplace problem -Delta u = 1 with
+//               homogeneous Dirichlet boundary conditions and, in this case,
+//               an inhomogeneous diffusion coefficient. The diffusion
+//               coefficient is given a small default value throughout the
+//               domain which is increased by two separate amounts in two named
+//               regions.
+//
+//               The example highlights the use of named attribute sets for
+//               both subdomains and boundaries in different contexts as well
+//               as basic methods to create named sets from existing attributes.
+
+#include "mfem.hpp"
+#include <fstream>
+#include <iostream>
+
+using namespace std;
+using namespace mfem;
+
+int main(int argc, char *argv[])
+{
+   // 1. Parse command-line options.
+   const char *mesh_file = "../data/compass.msh";
+   int order = 1;
+   string source_name = "Rose Even";
+   string ess_name = "Boundary";
+   bool visualization = true;
+
+   OptionsParser args(argc, argv);
+   args.AddOption(&mesh_file, "-m", "--mesh",
+                  "Mesh file to use.");
+   args.AddOption(&order, "-o", "--order",
+                  "Finite element order (polynomial degree) or -1 for"
+                  " isoparametric space.");
+   args.AddOption(&source_name,"-src","--source-attr-name",
+                  "Name of attribute set containing source.");
+   args.AddOption(&ess_name,"-ess","--ess-attr-name",
+                  "Name of attribute set containing essential BC.");
+   args.AddOption(&visualization, "-vis", "--visualization", "-no-vis",
+                  "--no-visualization",
+                  "Enable or disable GLVis visualization.");
+   args.Parse();
+   if (!args.Good())
+   {
+      args.PrintUsage(cout);
+      return 1;
+   }
+   args.PrintOptions(cout);
```

Comment:
@pazner why hasn't this simplification been made in existing examples like `ex1p`?

Reply:
I think it's just because of code churn — making the change would introduce new diffs that don't change functionality in any way. But this is a new example, so might as well use the newer way.